File: testdata/jennies/builders/map_of_builders/GoBuilder/map_of_builders/panel_builder_gen.go
package map_of_builders

import (
	cog "github.com/grafana/cog/generated/cog"
)

var _ cog.Builder[Panel] = (*PanelBuilder)(nil)

type PanelBuilder struct {
    internal *Panel
    errors cog.BuildErrors
}

func NewPanelBuilder() *PanelBuilder {
	resource := NewPanel()
	builder := &PanelBuilder{
		internal: resource,
		errors: make(cog.BuildErrors, 0),
	}

	return builder
}



func (builder *PanelBuilder) Build() (Panel, error) {
	if err := builder.internal.Validate(); err != nil {
		return Panel{}, err
	}
	
	if len(builder.errors) > 0 {
	    return Panel{}, cog.MakeBuildErrors("map_of_builders.panel", builder.errors)
	}

	return *builder.internal, nil
}

func (builder *PanelBuilder) Title(title string) *PanelBuilder {
    builder.internal.Title = title

    return builder
}

File: testdata/jennies/builders/map_of_builders/TypescriptBuilder/src/mapOfBuilders/panelBuilder.gen.ts
import * as cog from '../cog';
import * as mapOfBuilders from '../mapOfBuilders';

export class PanelBuilder implements cog.Builder<mapOfBuilders.Panel> {
    protected readonly internal: mapOfBuilders.Panel;

    constructor() {
        this.internal = mapOfBuilders.defaultPanel();
    }

    /**
     * Builds the object.
     */
    build(): mapOfBuilders.Panel {
        return this.internal;
    }

    title(title: string): this {
        this.internal.title = title;
        return this;
    }
}

